### DISCUSSION TITLE
VSK 5.0 Addon Request from Jharkhand Department

### DISCUSSION THREAD

**Discussion by manthanpandey23** [Sunbird-cQube/community#769] [Ideas]
We have an existing API to fetch data for Teacher Attendance and Student Attendance which give us data as per individual Teacher and Student level. So, we need a functionality to download these data as per date and page selected in excel format from the API in each of the module(Teacher and Student Attendance). So, they can download individual data for tracking.
Please find the attached Postman collection for testing.

[EY-Jharkhand.zip](https://github.com/Sunbird-cQube/community/files/11136304/EY-Jharkhand.zip)

**Comment by ananya-jain1** [comment]
Hi Manthan, currently, cQube's architecture doesn't support individual-level data in the visualisation layer. If the state wants to see the individual-level data on the cQube dashboards, the state can create these dashboards on an external visualisation tool (like, Metabase, Tableau etc) by connecting the available data. 

These dashboards, created on an external visualisation tool, can then be linked on cQube visualisation layer. Let me know in case you have any questions.

**Reply by manthanpandey23** [reply]
Hi Ananya, we are not looking for visualizing the data, we are requesting you to integrate functionality to download the data in excel format using state API attached.

**Reply by ananya-jain1** [reply]
Hey Manthan,
Let me explain what I understood from your response. 

You are suggesting that cQube should have a functionality where it consumes any state API with relevant data, reads it, converts it into an Excel Format so that it can be downloaded by the user. Is this correct? Do let me know if I am still missing anything.

**Reply by manish050994** [reply]
Hi Ananya,
Yes you are correct, 
we have 2 API's of student and teacher attendance we want to consume that API's through VSK to download Excel file.

**Reply by ananya-jain1** [reply]
Hi Manish,
Can you share the business need of this requirement? Who will use these files and how will they be used?

Additionally, this can also be done via Postman or any API tool right?

**Reply by manish050994** [reply]
Hi Ananya,

1. Yes, this can be done by Postman or any API tool but tools to handle API's are generally for developers and the calling team in client location is not so much tech-savvy and client wants one dashboard solution.
2.   **The Business Requirement** is to create one dashboard solution where they can see the visuals and download the report, as there is already a report in the dashboard but as we know client also want individual's tracking and for that we have to change the whole architecture of current system, so by integrating this API's to download reports they will get the desire report and we don't have to change the architecture also.
3. **Use of Report** is to track Individual's and calling team can also verify that who is present in the school and ask the District officers to verify the same.
